<?xml version="1.0"?>
<launch>
  <arg name="debug"  default="False" />  
  <arg name="blur_param_"			default="3" />
  <arg name="hsv_target_"			default="155" />
  <arg name="hsv_threshold_"	default="35" />
  <arg name="contour_area_min_"	default="100" />
  <arg name="contour_area_max_"	default="5000" />
  <arg name="contour_ratio_min_"	default="3" />
  <arg name="contour_ratio_max_"	default="9" />
  <arg name="contour_length_ratio_min_"	default="1" />
  <arg name="contour_length_ratio_max_"	default="4" />
  <arg name="aruco_detection_" 		default="False" />
  <arg name="image_topic_" 				default="/camera/rgb/image_rect_color" />
  <arg name="cloud_topic_"				default="/camera/depth/points" />
	
	<arg name="desc_match_thresh_"	default="0.005" />
	<arg name="gc_size_"						default="0.05"/>
	<arg name="gc_threshold_"				default="3"/>
		
  <node name="kinect_pose_estimation" pkg="marker_base_teaching" type= "kinect_pose_estimation" output="screen"> 
		<param name="blur_param_"          	type="int"   	value="$(arg blur_param_)"/> 
		<param name="hsv_target_"          	type="int"   	value="$(arg hsv_target_)"/> 
		<param name="hsv_threshold_"   			type="int"   	value="$(arg hsv_threshold_)"/> 
		<param name="contour_area_min_"    	type="int"   	value="$(arg contour_area_min_)"/> 
		<param name="contour_area_max_"     type="int"   	value="$(arg contour_area_max_)"/> 
		<param name="contour_ratio_min_"    type="double"   	value="$(arg contour_ratio_min_)"/> 
		<param name="contour_ratio_max_"    type="double"   	value="$(arg contour_ratio_max_)"/> 
		<param name="contour_length_ratio_min_"    type="double"   			value="$(arg contour_length_ratio_min_)"/> 
		<param name="contour_length_ratio_max_"    type="double"   			value="$(arg contour_length_ratio_max_)"/> 
		<param name="aruco_detection_"    	type="bool"   		value="$(arg aruco_detection_)"/> 
		<param name="image_topic_"					type="string"			value="$(arg image_topic_)"/>
		<param name="cloud_topic_"					type="string"			value="$(arg cloud_topic_)"/>
		<param name="gc_size_"							type="double"			value="$(arg gc_size_)"/>
		<param name="gc_threshold_"     		type="int"   	value="$(arg gc_threshold_)"/> 
		<param name="desc_match_thresh_"		type="double"			value="$(arg desc_match_thresh_)"/>
	</node>
	
</launch>


